### PR TITLE
minor fixes for calc-statistics

### PR DESCRIPTION
**Pull request by danish87**

### CODE REVIEW

**Review by shashankag14 — APPROVED** [review]

File: pcdet/models/detectors/pv_rcnn_ssl.py [hunk 1]
```diff
@@ -320,18 +320,18 @@ def forward(self, batch_dict):
...
-        for i in range(len(pred_boxes)):
-            valid_preds_mask = pred_boxes[i].sum(dim=-1) > 0
-            valid_gts_mask = gt_boxes[i].sum(dim=-1) > 0
+        for i in range(len(pred_boxes_org)):
+            valid_preds_mask = pred_boxes_org[i].sum(dim=-1) > 0
+            valid_gts_mask = gt_boxes_org[i].sum(dim=-1) > 0
             num_gts = valid_gts_mask.sum()
             num_preds = valid_preds_mask.sum()
             if num_gts > 0 and num_preds > 0:
-                gt_boxes = gt_boxes[i, valid_gts_mask]
```
Comment (fnozarian) [inline pcdet/models/detectors/pv_rcnn_ssl.py:333]: Can you please rename `gt_boxes` and `pred_boxes` to `valid_gt_boxes` and `valid_pred_boxes` instead of method arguments?

File: pcdet/models/detectors/pv_rcnn_ssl.py [hunk 2]
```diff
@@ -320,18 +320,18 @@ def forward(self, batch_dict):
...
     #  places during a batch.
-    def calc_statistics(self, pred_boxes, gt_boxes, update_metrics=True, tag=None, **kwargs):
-        assert len(pred_boxes) == len(gt_boxes)
-
+    def calc_statistics(self, pred_boxes_org, gt_boxes_org, update_metrics=True, tag=None, **kwargs):
+        assert len(pred_boxes_org) == len(gt_boxes_org)
+        
         statistics = defaultdict(list)
-        for i in range(len(pred_boxes)):
-            valid_preds_mask = pred_boxes[i].sum(dim=-1) > 0
```
Comment (fnozarian) [inline pcdet/models/detectors/pv_rcnn_ssl.py:328]: It turns out that this is not the correct way to filter empty (all-zero) rows, as the sum over columns could sometimes be negative!

File: pcdet/models/detectors/pv_rcnn_ssl.py [hunk 3]
```diff
@@ -341,15 +341,18 @@ def calc_statistics(self, pred_boxes, gt_boxes, update_metrics=True, tag=None, *
...
+                statistics['pseudo_fgs'].append(filtered_fg.mean().item())
+                statistics['pseudo_bgs'].append(filtered_bg.mean().item())
 
                 if kwargs['pseudo_sem_scores']:
                     pseudo_sem_scores = kwargs['pseudo_sem_scores']
-                    sem_score_fg = (pseudo_sem_scores[i] * (preds_iou_max > fg_thresh).float()).sum() \
-                                   / torch.clamp((preds_iou_max > fg_thresh).sum(), min=1.0)
-                    sem_score_bg = (pseudo_sem_scores[i] * (preds_iou_max < bg_thresh).float()).sum() \
-                                   / torch.clamp((preds_iou_max < bg_thresh).float().sum(), min=1.0)
+                    sem_score_fg = (pseudo_sem_scores[i][torch.nonzero(valid_preds_mask)] * filtered_fg).sum() \
```
Comment (fnozarian) [inline pcdet/models/detectors/pv_rcnn_ssl.py:352]: And it seems due to [incorrect valid mask](https://github.com/fnozarian/OpenPCDet/pull/10/files#r966239808), the `pseudo_sem_scores` does not match with the `filtered_fg` (which should match in general and there's no need to use `valid_pred_mask` for `pseudo_sem_scores`).

File: pcdet/models/detectors/pv_rcnn_ssl.py (hunk 2 above)
Comment (fnozarian) [reply pcdet/models/detectors/pv_rcnn_ssl.py:328]: Probably it can be `valid_preds_mask = pred_boxes[i].sum(dim=-1) != 0`.

Comment (shashankag14) [reply pcdet/models/detectors/pv_rcnn_ssl.py:328]: Using `valid_preds_mask = pred_boxes[i].sum(dim=-1) != 0` , we are summing all the elements of each `pred_boxes`. 

Eg. 
pred_boxes[i] = tensor([[  5.1124, -15.7919,  -0.7340,   4.9734,   2.0533,   2.1049,  -0.5451, 1.0000],
        [ 10.4329,   2.0733,  -0.6736,   4.7155,   1.8057,   1.7954,   1.0749, 1.0000],
        [ 15.8639,  -1.3333,  -0.6113,   4.1480,   1.7335,   1.7748,  -2.1551, 1.0000], device='cuda:0')

`pred_boxes[i].sum(dim=-1)` = [-1.8269, 22.2240, 20.4205]

But there could be a case when this sum is zero for a box, whereas the elements of the corresponding pred box are actually non-zero. To avoid that case, can we use `valid_preds_mask = pred_boxes[i][:, -1] != 0`

Comment (fnozarian) [reply pcdet/models/detectors/pv_rcnn_ssl.py:328]: Yeah I was thinking about the case when the sum can be zero even though the element,s are non-zero but it's really unlikely. That's why I said this is "probably" correct. I was also thinking about `valid_preds_mask = pred_boxes[i][:, -1] != 0` option. I think it's likely that the class labels are set to zero somewhere in the code or in the future. So checking this only based on the label might be risky. Having said that, a good way is maybe to check exactly for zero and reduce over `all` columns.
